code/datums/components/barber.dm

```diff
@@ -63,8 +62,30 @@ TYPEINFO(/datum/component/barber)
 	initialization_args = list()
 
 /datum/component/barber
+	var/list/all_hairs = list()
+	var/datum/appearanceHolder/new_AH
+	var/datum/movable_preview/character/preview
+	var/mob/living/carbon/human/barbee
+	var/mob/barber
+	var/hair_portion = "bottom"
+
 /datum/component/barber/Initialize()
 	. = ..()
+	var/all_hair_types = null
+
+	if (istype(src, /datum/component/barber/shave))
```

Comment:
if you're checking `istype(src)`, that means you should be moving the behaviour onto the child type (most of the time anyway)